tests/unit/llms/test_llm_loader.py

```diff
@@ -91,3 +108,29 @@ def mock_import(module, *args, **kwargs):
     with patch("builtins.__import__", side_effect=mock_import):
         with pytest.raises(Exception):
             LLMLoader(provider=provider, model=constants.GRANITE_13B_CHAT_V1)
+
+
+def _prepare_openapi_config():
+    providerConfig = ProviderConfig()
+    providerConfig.models = {constants.GRANITE_13B_CHAT_V1: "mock model"}
+    config.llm_config.providers = {constants.PROVIDER_OPENAI: providerConfig}
+
+
+def test_constructor_openai_llm_instance_no_api_key():
+    """Test the construction fo LLM instance for OpenAI when API key is not provided."""
+    _prepare_openapi_config()
+
+    # no API key is provided so validation should fail
+    with pytest.raises(Exception, match="Did not find openai_api_key"):
+        LLMLoader(
+            provider=constants.PROVIDER_OPENAI, model=constants.GRANITE_13B_CHAT_V1
+        )
+
+
+@patch.dict(os.environ, {"OPENAI_API_KEY": "key"})
```

Comment:
fyi this is going to need to be updated w/ @syedriko's changes which move us away from env based config loading and into yaml based config loading.